core/modules/nfv_ctrl.h

```diff
@@ -37,14 +38,19 @@ class NFVCtrl final : public Module {
 
  private:
   // All available per-core packet queues in a cluster
+  std::map<uint16_t, uint16_t> longTermOptimization(std::vector<double> flow_rate_per_bucket);
+  std::map<uint16_t, uint16_t> findMoves(double flow_rate_per_cpu[], std::vector<uint16_t> to_be_moved, std::vector<double> flow_rate_per_bucket);
```

Comment:
If not necessary, you can pass reference rather than pass copy.

Reply:
It makes sense to do that but can we add it as todo for now? It will require some careful changes as state of the data structures will have to be carefully managed and restored in failure cases.

Reply:
Okay to mark it todo. I don't get what you mean by managing for failure cases.
Do you mean that you call gurobi and do not get a result back?